pkg/services/ngalert/provisioning/provenance.go

```diff
@@ -0,0 +1,14 @@
+package provisioning
+
+import (
+	"github.com/grafana/grafana/pkg/services/ngalert/models"
+)
+
+func canUpdateWithProvenance(storedProvenance, provenance models.Provenance, errFun func() error) error {
+	isFileProvenance := storedProvenance == models.ProvenanceFile
+	isAPItoFileProvenance := storedProvenance == models.ProvenanceAPI && provenance == models.ProvenanceFile
+	if storedProvenance != provenance && (isFileProvenance || isAPItoFileProvenance) {
+		return errFun()
```

Comment:
I don't get why we need a callback function here. Why are you not using error wrapping instead, which is a much more idiomatic go? 

i.e.:
```go
err := canUpdateWithProvenance(a, b)
if err != nil {
  return fmt.Errorf("something: %w", err)
}
```

Reply:
I used a simpler approach. Thanks!